tests/test_zipimporter.py

```diff
@@ -0,0 +1,141 @@
+#------------------------------------------------------------------------------
+# Copyright (c) 2013, Nucleic Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#------------------------------------------------------------------------------
+import os
+import sys
+import enaml
+import struct
+import marshal
+import zipfile
+from enaml.core.parser import parse
+from enaml.core.enaml_compiler import EnamlCompiler
+from enaml.core.import_hooks import MAGIC, make_file_info
+from utils import wait_for_window_displayed
+
+
+def generate_cache(path):
+    #: Read
+    with open(path, 'rU') as f:
+        enaml_code = f.read()
+
+    #: Compile
+    ast = parse(enaml_code, filename=path)
+    code = EnamlCompiler.compile(ast, path)
+
+    #: Generate cache
+    with open('tmp.enamlc', 'wb') as f:
+        f.write(MAGIC)
+        f.write(struct.pack('i', int(os.path.getmtime(path))))
+        marshal.dump(code, f)
+    with open('tmp.enamlc', 'rb') as f:
+        data = f.read()
+    #: Cleanup
+    if os.path.exists('tmp.enamlc'):
+        os.remove('tmp.enamlc')
+
+    return data
+
+
+def make_library(lib):
+    # Create a library.zip with the examples.
+
+    with zipfile.ZipFile(lib, 'w') as zf:
+        for src, dst in [
+                          ('examples/widgets/buttons.enaml', 'buttons.enaml'),
+                          (None, 'package/__init__.py'),
+                          ('examples/widgets/form.enaml', 'package/form.enaml'),
+                          (None, 'package/subpackage/__init__.py'),
+                          ('examples/widgets/slider.enaml', 'package/subpackage/slider.enaml'),
+
+                         ]:
+            if src:
+                zf.write(src, dst, compress_type=zipfile.ZIP_DEFLATED)
+            else:
+                #: Create the __init__.py
+                zf.writestr(dst, '', compress_type=zipfile.ZIP_DEFLATED)
+
+        #: Generate cache for splitter and notebook
+        for src, dst in [
+                ('examples/widgets/splitter.enaml',
+                    make_file_info('splitter.enaml').cache_path),
+                ('examples/widgets/notebook.enaml',
+                    make_file_info('package/subpackage/notebook.enaml').cache_path)
+            ]:
+            zf.writestr(dst, generate_cache(src), compress_type=zipfile.ZIP_DEFLATED)
+
+
+def with_library(f):
```

Comment:
This could be turned into a pytest fixture. I can do it if you are not familiar with them.

Reply:
Yes that would be good.